src/platforms/hosted/ftdi_bmp.c

```diff
@@ -379,6 +379,100 @@ const cable_desc_s cable_desc[] = {
 	{},
 };
 
+/*
+	Search the adapter descriptor table for probes match the VID/PID
+	in the given probe structure.
+	If a single match is found, place the adapter descriptor pointer into
+	the cl_opts structure and return true.
+
+	Otherwise return false.
+ */
+
+bool ftdi_lookup_adapter_from_vid_pid(const probe_info_s *probe, bmda_cli_options_s *cl_opts)
+{
+	size_t adapter_count = 0;
+	const cable_desc_s *cable = &cable_desc[0];
+	const cable_desc_s *selection = NULL;
+
+	//
+	// If the user entered a serial number, check if the attached probe
+	// is the right one
+	//
+	if (cl_opts->opt_serial) {
+		if (strstr(cl_opts->opt_serial, probe->serial)) {
+			adapter_count = 1;
+		}
+	}
+	if (!cl_opts->opt_cable && !adapter_count) {
+		while (cable->vendor) {
+			if (cable->vendor == probe->vid && cable->product == probe->pid) {
+				// bool found = false;
+				// /*
+				// 	If the user entered a probe name, check if this is the one
+				//  */
+				// if (cl_opts->opt_cable != NULL) {
+				// 	if (strstr(cl_opts->opt_cable, cable->name)) {
+				// 		found = true;
+				// 	}
+				// }
+				// else if(strstr(cl_opts->opt_serial, probe->serial)) {
+				// 	found = true ;
+				// }
+				adapter_count++;
+				selection = cable;
+				// if(found)
+				// {
+				// 	break ;
+				// }
+			}
+			cable++;
+		}
+		if (adapter_count == 1) {
+			/* place the adapter name into cl_opts */
+			cl_opts->opt_cable = selection->name;
+		}
+	} else {
+		adapter_count = 1;
+	}
+	return adapter_count == 1;
+}
+
+bool ftdi_lookup_cable_by_product(bmda_cli_options_s *cl_opts, const char *product)
+{
+	bool result = false;
+	const cable_desc_s *cable = &cable_desc[0];
+	if (!cl_opts->opt_cable) {
```

Comment:
Given this structure:
```suggestion
	if (cl_opts->opt_cable)
		return false;
```
so the entire rest of the function can be de-indented, simplifying the control flow